### ISSUE TITLE
wp rocket and other recommendations

### ISSUE THREAD

**Issue by peixotorms** [GoogleChrome/lighthouse#14894]
This is related to `https://pagespeed.web.dev/`
Just cross referring to `https://github.com/GoogleChromeLabs/psi/issues/134` as I am not 100% sure which repository is authoritative for this.

> Hi,
> 
> I don't think it's ethical to promote paid products like wp rocket via the pagespeed insights tests.
> `https://i.imgur.com/dOFEeE3.png`
> 
> At most, you could promote open source, free plugins from the wordpress directory, but even that should be considered, as it offers a significant advantage to have 'google's pagespeed" recognition, vs not having it.
> 
> Unless WP Rocket is owned by Google or it's free open source (it is not), it should never show up on the list of recommendations.
> 
> It doesn't matter if it's a good product or not.
> You are directly favoring a paid product over anything else, simply by listing it.

**Comment by Generosus** [comment]
We noticed the same. We second @peixotorms's concern. Advertising WP-Rocket to help solve PSI issues is not the right way to go. Please remove it. Thank you.

**Comment by paulirish** [comment]
Hi yall. Good to hear from you again Raul. We always appreciate your bug reports.

These little recommendations come from "stack packs".  https://github.com/GoogleChrome/lighthouse-stack-packs
They ONLY show up if the specific technology is detected. So.. The page that [this report](https://i.imgur.com/dOFEeE3.png) is for was already using WP Rocket.   We use `js-library-detector` for detection; here's [its wp-rocket detection](https://github.com/johnmichel/Library-Detector-for-Chrome/blob/3135bb24072af715f45c8c54232c7a96d11b7b09/library/libraries.js#L1933-L1947).

We 100% agree with you that Lighthouse should not be advertising paid products. We should not upsell folks. For that reason we've actually shut down a few folks who had a freemium product and wanted a stackpack; there was a chance they'd recommend paid solutions to users using the free thing. No bueno. And Raul, I also prefer folks gravitate towards awesome open-source solutions. 

One implication... If WP-Rocket ever _adds_ a free offering.... And we cannot differentiate it from the paid thing, then we'd have to disable their stackpack.

If there's a bug in the wp-rocket detection in js-lib-detector.. That's a problem and we'd definitely work to get that fixed... But it looks pretty fine to me. 

Lastly, anyone can [propose a new stackpack](https://github.com/GoogleChrome/lighthouse-stack-packs/blob/master/CONTRIBUTING.md#submitting-a-new-stack-pack). Open-source or paid or whatever. (We don't have the stipulations about freemium noted in that doc, but we should). 

---------

@alexnj had an idea for making this situation more clear. Perhaps we can add a message of `This page seems to be using <technology>:`.  

What do you think about a tooltip on the logo that says "This page is using WP Rocket 4.3.2" or "This page is using WordPress 6.1.1"  ?

**Comment by Generosus** [comment]
Hey @paulirish,

Great input. So you know, unless we missed something, the recommendation to use WP Rocket appears whether using the technology (plugin) or not. We don't use it, yet it appears (or appeared) at the end of our PSI/LH tests.

Hey @alexnj,

Good suggestion, but quite frankly, best to leave all commercial products out of PSI/LH -- including WP Rocket. This takes us back to @peixotorms's point.

**Comment by paulirish** [comment]
Can you share the URL of the page? We need to investigate why its being detected if its not there.

**Comment by Generosus** [comment]
Hey @paulirish,

Had a gut feeling you were going to come back with that request :)

Allow us to replicate the issue and we'll provide the steps/info you requested.

We did use WP Rocket at one time, so not sure if Lighthouse Stack Packs is detecting "old" WP Rocket data stored in our database(s). Our website(s) are running clean at this time, so we'll need to simulate a couple of failures in order to obtain the corresponding PSI/LH recommendations referencing WP Rocket. 

Please don't let us distract you (or other commenters) from @peixotorms's main point.

Stay tuned.

**Comment by peixotorms** [comment]
@paulirish thank you for the clarification. If the recommendation only shows if a site is already using a certain product, then it "can" be acceptable to me. I checked on that specific site that wp rocket was indeed installed, and then tested another one without, and the recommendation didn't show up.

I checked the stack path and it appears to be detecting it fine, so for me, we can close this.

I see now that the purpose of the recommendation is to help you improve an existing product, though I am not sure about the utility of this. Several of the recommendations on wp rocket break the design on several sites when enabled, so if a user is simply going to enable them because pagespeed tells them to, their site layout might break.

Of course, they can always disable it, but this is akin of extending product support through pagespeed insights, so I would still not include it. 

As a paid product they could very well advertise that they are in partnership with the pagespeed test, or that they integrate with it, and it would not be 100% false. Even if they don't do, influencers and other people might blog about it. 
That being said, it's still fair, as long as anyone can do the same, so no complains here.
